File: data/src/main/java/com/balv/imdb/data/model/MovieEntity.java
package com.balv.imdb.data.model;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "movies")
public class MovieEntity {
    @PrimaryKey
    @NonNull
    public String id;
    public String title;
    public String released;
    public String time;
    public String genre;
    public String director;
    public String writer;
    public String actors;
    public String poster;
    public String imdbRated;
    public String plot;
}

File: data/src/main/java/com/balv/imdb/data/model/MovieEntity.kt
package com.balv.imdb.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "movies")
data class MovieEntity(
    @PrimaryKey
    val id: String,
    val title: String? = null,
    val released: String? = null,
    val time: String? = null,
    val genre: String? = null,
    val director: String? = null,
    val writer: String? = null,
    val actors: String? = null,
    val poster: String? = null,
    val imdbRated: String? = null,
    val plot: String? = null
)
